lib/rstreamor/response.rb

```diff
@@ -1,25 +1,25 @@
 module Rstreamor
   class Response
-    attr_accessor :request
+    attr_reader :request
 
     def initialize(request)
-      self.request = request
+      @request = request
     end
 
     def response_code
-      self.request.range_header? ? 206 : 200
+      request.range_header? ? 206 : 200
     end
 
     def content_length
-      if self.request.range_header?
-        (self.request.upper_bound - self.request.lower_bound).to_s
+      if request.range_header?
+        (request.upper_bound - request.lower_bound).to_s
       else
-        self.request.file.data.size
+        request.file_size
       end
     end
 
     def content_range
-      "bytes #{self.request.lower_bound}-#{self.request.upper_bound - 1}/#{self.request.file.data.size}"
+      "bytes #{ request.lower_bound }-#{ request.upper_bound - 1 }/#{ request.file_size }"
```

Comment:
Here we make sure of using the size we are calculating in `File` through `Request`'s proxy, instead of recalculating.